Migrate ci to git hub actions

gotanygrapes-lemonade opened the pull request:
### Context

We have [a sensible default](https://docs.envato.net/architecture-guild/sensible-defaults/expected/open-source-ci.html) to **not** use our internal Buildkite agents to run CI for open-source projects.

However, we're doing that for a number of open-source Buildkite plugins. (Probably historical, before the default was in-place.) Let's update these projects to use GitHub Actions.


### Considerations

An [example GitHub Actions configuration](https://github.com/envato/heroku-container-deploy-buildkite-plugin/blob/16f73f647bbb6e5cca4217ec5413e6e3d3bb2f41/.github/workflows/test.yml) we can use for inspiration (or copy).

- Adding a YAML file to the `.github/workflows/` directory should be enough to enable GitHub actions
- Delete Buildkite pipelines by navigating to their settings page in a browser and click the `Delete Pipeline` in the pipeline management section

Commit:
add test.yml file to use GitHub Actions

Commit:
Update the badge in README with Github Actions Badge

Commit:
remove .buildkite directory

orien reviewed (commented):
Excellent. Thank you.

Commit:
apply Orien's suggestions:

- put MIT License badge back in README
- change tests.yml to run bats tests in the mounted volume directory

Commit:
modify tests.yml to run tests on pull request

orien commented on README.md:
```diff
@@ -1,6 +1,7 @@
 
-![Build status](https://badge.buildkite.com/3e169bf223b02ed62d3a3090e2ef78cb025e83076c33b694b5.svg?branch=main)
+[![tests](https://github.com/envato/bundle-update-buildkite-plugin/actions/workflows/tests.yml/badge.svg?branch=main)](https://github.com/envato/bundle-update-buildkite-plugin/actions/workflows/tests.yml)
+
```
Let's remove this empty line so the badges end up on the same line when rendered. It currently looks like this:

<img width="486" alt="image" src="https://user-images.githubusercontent.com/497874/152704595-616c0217-54be-44ea-ac2c-bfd46cb7b428.png">

```suggestion
```

orien reviewed (approved):
Nice one 👍

gotanygrapes-lemonade committed:
Update README.md

Co-authored-by: Orien Madgwick <_@orien.io>
